pkg/operator/certrotation/cabundle.go

```diff
@@ -98,6 +100,10 @@ func (c CABundleConfigMap) EnsureConfigMapCABundle(ctx context.Context, signingC
 		caBundleConfigMap = actualCABundleConfigMap
 	} else if updateRequired {
 		actualCABundleConfigMap, err := c.Client.ConfigMaps(c.Namespace).Update(ctx, caBundleConfigMap, metav1.UpdateOptions{})
+		if err != nil && strings.Contains(err.Error(), genericregistry.OptimisticLockErrorMsg) {
```

Comment:
I'm wary of doing this based on an error message substring because of the common practice of blindly concatenating error messages. Are you worried about existing or potential future cases where https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#IsConflict returns true but the error is not a resourceVersion mismatch?

Reply:
Since we only use SSA for these secrets I think any conflict would be resourceVersion mismatch iiuc

Reply:
Testing `apierrors.IsConflict` in https://github.com/openshift/cluster-kube-apiserver-operator/pull/1819